blendergltf.py

```diff
@@ -946,29 +946,51 @@ def png_pack(png_tag, data):
         png_pack(b'IDAT', zlib.compress(raw_data, 9)),
         png_pack(b'IEND', b'')])
 
-    return 'data:image/png;base64,' + base64.b64encode(png_bytes).decode()
+    if bytes:
+        return png_bytes
+    else:
+        return 'data:image/png;base64,' + base64.b64encode(png_bytes).decode()
 
 
 def export_images(settings, images):
     def check_image(image):
         errors = []
         if image.size[0] == 0:
-            errors.add('x dimension is 0')
+            errors.append('x dimension is 0')
         if image.size[1] == 0:
-            errors.add('y dimension is 0')
+            errors.append('y dimension is 0')
+        if image.type != 'IMAGE':
+            errors.append('not an image')
 
-        if errors:
+        if len(errors) > 0:
             err_list = '\n\t'.join(errors)
             print('Unable to export image {} due to the following errors:\n\t{}'.format(image.name, err_list))
             return False
 
         return True
 
+    extMap = {'BMP': 'bmp', 'JPEG': 'jpg', 'PNG': 'png', 'TARGA': 'tga'}
     def export_image(image):
         uri = ''
 
         storage_setting = settings['images_data_storage']
-        if storage_setting == 'COPY':
+        image_packed = image.packed_file != None
+        if image_packed and storage_setting in ['COPY','REFERENCE']:
```

Comment:
When a file is packed, there is no "place" to reference them. If we don't export them, they don't exist.